### PR TITLE
FIX_bdd: création de la constante domaine studyfield et du select studyfield

### PR DESCRIPTION
**Pull request by maxime-guichard**

### CODE REVIEW

**Review by Erdnaxela45 — DISMISSED** [review]

File: src/DataFixtures/JobFixtures.php [hunk 1]
```diff
@@ -27,7 +28,9 @@ public function load(ObjectManager $manager)
             $job->setCity($faker->city());
             $job->setDescription($faker->paragraph(3, true));
             $job->setCompany($this->getReference('company_' . rand(1, CompanyFixtures::LOOPNUMBER)));
-            $job->setStudyField($this->getReference('studyField_' . rand(1, StudyFieldFixtures::LOOPNUMBER)));
+            for ($i = 1; $i < count(StudyFieldFixtures::STUDYFIELD); $i++) {
+                $job->setStudyField($this->getReference('studyField_' . $i));
+            }
```
Comment (LucyMerlier) [inline src/DataFixtures/JobFixtures.php:33]: this just ends up giving the same study field to all your jobs => do a random like for the company?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comment

**Review by endricklosange — APPROVED** [review]

File: src/DataFixtures/JobFixtures.php [hunk 2]
```diff
@@ -27,7 +28,11 @@ public function load(ObjectManager $manager)
             $job->setCity($faker->city());
             $job->setDescription($faker->paragraph(3, true));
             $job->setCompany($this->getReference('company_' . rand(1, CompanyFixtures::LOOPNUMBER)));
-            $job->setStudyField($this->getReference('studyField_' . rand(1, StudyFieldFixtures::LOOPNUMBER)));
+            for ($i = 0; $i < count(StudyFieldFixtures::STUDYFIELD); $i++) {
+                $job->setStudyField(
+                    $this->getReference('studyField_' . rand(1, count(StudyFieldFixtures::STUDYFIELD) - 1))
+                );
```
Comment (LucyMerlier) [inline src/DataFixtures/JobFixtures.php:34]: this rand should start at 0, and should not be inside a loop => just delete lines 31 and 35

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comment

**Review by Erdnaxela45 — APPROVED** [review]

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
all this is also done in [this PR](https://github.com/WildCodeSchool/orleans-202103-php-project-interim/pull/78)

**Pull request closed without merging**